python-checks/src/test/resources/checks/datetime_constructor_use_timezone_aware.py

```diff
@@ -0,0 +1,40 @@
+def some_function():
+    from datetime import datetime
+    datetime.utcnow() # Noncompliant {{Using timezone aware "datetime"s should be preferred over using "datetime.datetime.utcnow" and "datetime.datetime.utcfromtimestamp"}}
+   #^^^^^^^^^^^^^^^^^
+    timestamp = 1571595618.0
+    datetime.utcfromtimestamp(timestamp) # Noncompliant {{Using timezone aware "datetime"s should be preferred over using "datetime.datetime.utcnow" and "datetime.datetime.utcfromtimestamp"}}
+   #^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+def other_function():
+    import datetime
+    timestamp = 1571595618.0
+    datetime.datetime.utcnow() # Noncompliant
+    datetime.datetime.utcfromtimestamp(timestamp) # Noncompliant
+
+    from datetime import datetime as dt
+    dt.utcnow() # Noncompliant
+    dt.utcfromtimestamp(timestamp) # Noncompliant
+
+def unknown_symbols():
+
+    unrelated_call()
+
+    datetime.datetime.utcnow() # FN because there is no symbol
+    datetime.datetime.utcfromtimestamp() # FN because there is no symbol
+
+def compliant_examples():
+    from datetime import datetime, timezone
+
+    datetime.now(timezone.utc)
+    timestamp = 1571595618.0
+    datetime.fromtimestamp(timestamp, timezone.utc)
+
+def aliased_utcnow():
+    from datetime import datetime
+    aliased = datetime.utcnow
+    # FN because we lose the FQN in the assignment
+    aliased()
```

Comment:
This is quite nitpicky but to me, aliased would be something like:
```
from datetime.datetime import utcnow as aliased
aliased() # Noncompliant
```
Which we'd most likely raise correctly.
Here the symbol is reassigned (and it's indeed FN in that case, good thing to document it!)